### PR TITLE
Allow group admins to edit group exercises

### PR DESCRIPTION
**Pull request by Teyras**

### CODE REVIEW

File: app/model/entity/Group.php [hunk 1]
```diff
@@ -306,6 +306,24 @@ public function isAdminOrSupervisorOfSubgroup(User $user): bool {
     return false;
   }
 
+  /**
+   * User is admin of a group or any of its parents
+   * @param User $user
+   * @return bool
+   */
+  public function isAdminOfSuperGroup(User $user): bool {
+    $group = $this;
```
Comment (Neloop) [inline app/model/entity/Group.php]: There is no need to go through all parent groups again... there is `getAdminsIds()` method

File: app/config/permissions.neon [hunk 2]
```diff
@@ -306,6 +306,23 @@ permissions:
       conditions:
           - exercise.isAuthor
 
+    - allow: true # Admin of the group of an exercise has most of the rights to the exercise
```
Comment (Neloop) [inline app/config/permissions.neon]: Wouldn't it be better to add condition `exercise.isSuperGroupAdmin` to above `exercise.isAuthor` and separate `delete` actions into new permission rule?

Comment (Teyras) [reply app/config/permissions.neon]: Conditions are joined with `&&`, so this wouldn't work. I still welcome suggestions for a good `||` syntax, though.

File: app/model/entity/Group.php (hunk 1 above)
Comment (Teyras) [reply app/model/entity/Group.php]: That would work too. Do you think it's worth rewriting it?

File: app/config/permissions.neon (hunk 2 above)
Comment (Neloop) [reply app/config/permissions.neon]: I though something like `exercise.isAuthorOrSuperGroupAdmin` method... but not sure if it is a good idea

File: app/model/entity/Group.php (hunk 1 above)
Comment (Neloop) [reply app/model/entity/Group.php]: Well... I think so, if I am not mistaken `isAdminOf()` method is using `getAdminsIds()` method so... you are doing searching which is not really necessary...

File: app/config/permissions.neon (hunk 2 above)
Comment (Teyras) [reply app/config/permissions.neon]: That sounds way too specific for my taste.

Comment (Neloop) [reply app/config/permissions.neon]: Ok, fair enough...

File: app/model/entity/Group.php (hunk 1 above)
Comment (Teyras) [reply app/model/entity/Group.php]: You mean `getAdminIds` doesn't do the searching under the hood?

Comment (Neloop) [reply app/model/entity/Group.php]: Look in the code please...

**Pull request merged**